File: tracks_reconstruction/4_MainRecSof/EvaluateAngularCoordinates.cc
#include "EvaluateAngularCoordinates.h"
#include "TMath.h"

using namespace std; 
vector<double> TrackAngularCoordinates(double slope_xy, double slope_xz, double X0, double X2) {

  double Dz,Dx,Dy;
  double theta=0, phi=0;
  vector<double> coordinates;
  Dz = slope_xz*(X0 - X2);
  Dx = X2 - X0;
  Dy = slope_xy*Dx;
  theta = atan(abs(Dz)/(sqrt(Dx*Dx +Dy*Dy)))*(180./TMath::Pi());
  
  if(Dz > 0) {
    phi = atan(slope_xy)*(180./TMath::Pi()) +180.;
  }
  else {
    if(slope_xy>0) phi = atan(slope_xy)*180/TMath::Pi();
    else phi = 360. + 	atan(slope_xy)*180/TMath::Pi();             
    
  }
  
  coordinates.push_back(theta);
  coordinates.push_back(phi);
  return coordinates;
}

  

File: muraves/tracks_reconstruction/4_MainRecSof/evaluate_angular_coordinates.py
from __future__ import annotations

import math


def TrackAngularCoordinates(slope_xy: float, slope_xz: float, X0: float, X2: float) -> list[float]:
    """Python port of TrackAngularCoordinates from EvaluateAngularCoordinates.cc."""
    dz = slope_xz * (X0 - X2)
    dx = X2 - X0
    dy = slope_xy * dx

    theta = math.atan(abs(dz) / math.sqrt(dx * dx + dy * dy)) * (180.0 / math.pi)

    if dz > 0:
        phi = math.atan(slope_xy) * (180.0 / math.pi) + 180.0
    else:
        if slope_xy > 0:
            phi = math.atan(slope_xy) * 180.0 / math.pi
        else:
            phi = 360.0 + math.atan(slope_xy) * 180.0 / math.pi

    return [theta, phi]


# Pythonic alias
track_angular_coordinates = TrackAngularCoordinates
